### PR TITLE
test(devcontainer/ci): Test on x86 and on arm

### PR DESCRIPTION
**Pull request by lmilbaum**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: .github/workflows/devcontainer.yml [hunk 1]
```diff
@@ -7,11 +7,18 @@ on:
...
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2.1.0
+
       - name: Build and run dev container task
         uses: devcontainers/ci@57eaf0c9b518a76872bc429cdceefd65a912309b # v0.3.1900000329
         with:
-          runCmd: yarn build
+          runCmd: yarn test
```
Comment (viceice) [inline .github/workflows/devcontainer.yml:23]: no, test is too slow. we waste to much time when running tests again here

**Review by viceice — CHANGES_REQUESTED** [review]

File: .github/workflows/devcontainer.yml (hunk 1 above)
Comment (lmilbaum) [reply .github/workflows/devcontainer.yml:23]: Won't it make more sense to run the tests only here or maybe consolidate the current build workflow  with the devcontainer one?

Comment (viceice) [reply .github/workflows/devcontainer.yml:23]: no, tests are already very slow and we don't want them to get slower. arm64 emulation would be magnitudes slower, so tests would probably need hours to run

Comment (lmilbaum) [reply .github/workflows/devcontainer.yml:23]: Yup. You are correct. I've checked the time it took for the devcontainer workflow to run -> 45/46 minutes while the regular one took 19 minutes.
I just wonder if `yarn build` is enough for exposing the ARM related errors. How about running a small subset of the unit tests?

File: .github/workflows/devcontainer.yml [hunk 2]
```diff
@@ -7,11 +7,18 @@ on:
...
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2.1.0
+
       - name: Build and run dev container task
         uses: devcontainers/ci@57eaf0c9b518a76872bc429cdceefd65a912309b # v0.3.1900000329
         with:
           runCmd: yarn build
+          platforms: linux/amd64, linux/arm64
```
Comment (JamieMagee) [inline .github/workflows/devcontainer.yml:24]: There's a warning in the build:

> Warning: Unexpected input(s) 'platforms', valid inputs are ['imageName', 'imageTag', 'platform', 'runCmd', 'subFolder', 'checkoutPath', 'push', 'refFilterForPush', 'eventFilterForPush', 'env', 'skipContainerUserIdUpdate', 'userDataFolder', 'cacheFrom', 'noCache']

Comment (JamieMagee) [inline .github/workflows/devcontainer.yml:24]:
```suggestion
          platform: linux/amd64, linux/arm64
```

File: .github/workflows/devcontainer.yml [hunk 3]
```diff
@@ -7,11 +7,18 @@ on:
 jobs:
   devcontainer-test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [arm64, amd64]
```
Comment (viceice) [inline .github/workflows/devcontainer.yml:12]:
```suggestion
```
isn't used and running twice makes no sense

File: .github/workflows/devcontainer.yml [hunk 4]
```diff
@@ -7,11 +7,18 @@ on:
...
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2.1.0
+
       - name: Build and run dev container task
         uses: devcontainers/ci@57eaf0c9b518a76872bc429cdceefd65a912309b # v0.3.1900000329
         with:
           runCmd: yarn build
+          platform: linux/amd64, linux/arm64
```
Comment (JamieMagee) [inline .github/workflows/devcontainer.yml:24]: > ERROR: " linux" is an invalid component of " linux/arm64": platform specifier component must match "^[A-Za-z0-9_-]+$": invalid argument

```suggestion
          platform: linux/amd64,linux/arm64
```

File: .github/workflows/devcontainer.yml [hunk 5]
```diff
@@ -11,7 +11,11 @@ jobs:
       - name: Checkout
         uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
 
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2.1.0
```
Comment (viceice) [inline .github/workflows/devcontainer.yml:15]:
```suggestion
      - name: Set up Docker
        uses: containerbase/internal-tools@1b85c9fe3081ccde537e99a5e0d47fa074119bdf # v1.17.3
        with:
          command: docker-config
```

**Comment by viceice** [comment]
It's now building both platforms, but still only testing the x86_64 image

**Comment by lmilbaum** [comment]
Closing the PR after reading [doc](https://github.com/devcontainers/ci/blob/3514d191cb2a190a133be85abcd83a868e2281d3/docs/multi-platform-builds.md?plain=1)

**Pull request closed without merging**